### PR TITLE
[#10969] E2E tests for Audit Logs

### PR DESCRIPTION
**Pull request by AdithyaNarayan**
Part of #10969 

E2E tests for the following
- [x] Default date and time
- [x] Empty logs when no record is present
- [x] Log change after submission (this is not completed as masquerade mode does not work on staging server but logs only work on staging)

Any help for the third test case would be appreciated.

### CODE REVIEW

File: src/e2e/java/teammates/e2e/pageobjects/InstructorAuditLogsPage.java
Comment (t-cheepeng) [inline src/e2e/java/teammates/e2e/pageobjects/InstructorAuditLogsPage.java:58]: I'm assuming this method retrieves all the logs that are being displayed? In that case, this will never collect the data if there were logs present since you find by `tagname "i"`. The only possibility of that is when there is no logs

File: src/e2e/java/teammates/e2e/cases/InstructorAuditLogsPageE2ETest.java
Comment (t-cheepeng) [inline src/e2e/java/teammates/e2e/cases/InstructorAuditLogsPageE2ETest.java:69]: You can probably look into stubbing the LocalLoggingService to return some logs that you then use to verify

**Review by t-cheepeng — CHANGES_REQUESTED** [review]

File: src/e2e/java/teammates/e2e/cases/InstructorAuditLogsPageE2ETest.java
Comment (daongochieu2810) [reply src/e2e/java/teammates/e2e/cases/InstructorAuditLogsPageE2ETest.java:69]: Would doing this defeat the purpose of blackbox testing?

File: src/e2e/java/teammates/e2e/pageobjects/StudentFeedbackSubmissionPage.java
Comment (daongochieu2810) [inline src/e2e/java/teammates/e2e/pageobjects/StudentFeedbackSubmissionPage.java:31]: is this try-catch necessary here?

File: src/e2e/resources/data/InstructorAuditLogsPageE2ETest.json
Comment (daongochieu2810) [inline src/e2e/resources/data/InstructorAuditLogsPageE2ETest.json:2]: remove the data that you did not use

**Review by daongochieu2810 — CHANGES_REQUESTED** [review]

File: src/e2e/java/teammates/e2e/cases/InstructorAuditLogsPageE2ETest.java
Comment (t-cheepeng) [reply src/e2e/java/teammates/e2e/cases/InstructorAuditLogsPageE2ETest.java:69]: That is true. But I don't think gcloud has logging emulators right now. 
https://github.com/googleapis/google-cloud-java/blob/master/TESTING.md#testing-code-that-uses-logging

Alternatives include a remote testing project but that is likely unfeasible. So next best alternative is to have greybox testing. Any ideas @halfwhole ?

**Comment by AdithyaNarayan** [comment]
@t-cheepeng @daongochieu2810 I updated the test as suggested

**Comment by daongochieu2810** [comment]
It seems like there is a NullPointerException for this e2e test. My guess is that you are trying to get a non-existent session from the HashMap in AuditLogPage

**Comment by AdithyaNarayan** [comment]
Apparently, the `NullPointerException` was from a typo in the test data which I forgot to fix while linting. Fixed 👍

**Comment by daongochieu2810** [comment]
@AdithyaNarayan any update on this?

**Comment by AdithyaNarayan** [comment]
It should be ready to review now. Not sure why but even after merging with master, the timezone database was not updated. Had to do it manually

**Comment by daongochieu2810** [comment]
@AdithyaNarayan Please remove changes from package-lock.json

**Comment by AdithyaNarayan** [comment]
Done!

**Comment by daongochieu2810** [comment]
@AdithyaNarayan Can you please explain the need of updating `TimezoneSyncerTest`? Also please address my previous comments

**Comment by AdithyaNarayan** [comment]
The frontend and backend timezone versions were out of sync. I followed the instructions given in the [maintainer guide](https://github.com/TEAMMATES/teammates-ops/blob/master/maintainer-guide.md#timezone-database-update) to update both to 2021a. I have also replaced the try-catch block.

**Review by daongochieu2810 — APPROVED** [review]
lgtm

**Comment by madanalogy** [comment]
> The frontend and backend timezone versions were out of sync. I followed the instructions given in the [maintainer guide](https://github.com/TEAMMATES/teammates-ops/blob/master/maintainer-guide.md#timezone-database-update) to update both to 2021a. I have also replaced the try-catch block.

You should be doing this in a separate PR. Open a new issue indicating that the timezone's gone wonky then submit a PR to fix that first. If I recall, @halfwhole did something like that before. Just for future note. 

Also I think the tests are working now so you can revert those changes before we proceed with this PR.

**Comment by AdithyaNarayan** [comment]
Reverted!

**Review by madanalogy — APPROVED** [review]

**Review by daongochieu2810 — APPROVED** [review]

**Review by t-cheepeng — APPROVED** [review]

**Pull request merged**
